### PR TITLE
Infer loader from filetype

### PR DESCRIPTION
**Pull request by ifahrentholz**
== Description ==
The pv.config.js should be as minimal as possible. To accomplish that we can infer some config values from the filetype. For example if the user configures the jsEntry to be "src/index.tsx" we can automatically set "useTS" and "useReact" to be true.

== Closes issue(s) ==
#39

== Changes ==
paths.js => config will be piped through the inferLoaderFromFiletype function.

== Affected Packages ==
webpack-config

### CODE REVIEW

File: packages/assemble-lite/package.json [hunk 1]
```diff
@@ -21,9 +21,7 @@
     "type": "git",
     "url": "git+https://github.com/pro-vision/fe-tools.git"
   },
-  "scripts": {
-    "test": "echo \"Error: run tests from root\" && exit 1"
-  },
+  "scripts": {},
```
Comment (friewerts) [inline packages/assemble-lite/package.json:24]: i think this key can be removed

File: packages/pv-scripts/package.json [hunk 2]
```diff
@@ -19,9 +19,7 @@
     "type": "git",
     "url": "https://github.com/pro-vision/fe-tools"
   },
-  "scripts": {
-    "test": "echo \"Error: run tests from root\" && exit 1"
-  },
+  "scripts": {},
```
Comment (friewerts) [inline packages/pv-scripts/package.json:22]: i think this key can be removed

File: packages/pv-stylemark/package.json [hunk 3]
```diff
@@ -22,9 +22,7 @@
     "type": "git",
     "url": "https://github.com/pro-vision/fe-tools"
   },
-  "scripts": {
-    "test": "echo \"Error: run tests from root\" && exit 1"
-  },
+  "scripts": {},
```
Comment (friewerts) [inline packages/pv-stylemark/package.json:25]: i think this key can be removed

File: packages/webpack-config/babel.config.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+// babel.config.js
```
Comment (friewerts) [inline packages/webpack-config/babel.config.js:1]: i think all the jest-related stuff should be located in the root dir, to be later extended by the packages

File: packages/webpack-config/jest.config.js [hunk 5]
```diff
@@ -0,0 +1,187 @@
+// For a detailed explanation regarding each configuration property, visit:
```
Comment (friewerts) [inline packages/webpack-config/jest.config.js:1]: i think all the jest-related stuff should be located in the root dir, to be later extended by the packages

File: packages/webpack-config/package.json [hunk 6]
```diff
@@ -66,5 +65,11 @@
     "style-loader": "1.0.1",
     "webpack": "4.41.2",
     "webpack-merge": "4.2.2"
+  },
+  "devDependencies": {
```
Comment (friewerts) [inline packages/webpack-config/package.json:69]: i think all the jest-related stuff should be located in the root dir, to be later extended by the packages

File: packages/webpack-config/src/helpers/getFileExtension/getFileExtension.js [hunk 7]
```diff
@@ -0,0 +1,11 @@
+/**
+ * Get the file extension from a filepath or filename.
+ *
+ * @param {string} file - filename or filepath
+ * @returns {string} file extension
+ */
+function getFileExtension(file) {
+  return file.split(".").reverse()[0];
```
Comment (friewerts) [inline packages/webpack-config/src/helpers/getFileExtension/getFileExtension.js:8]: maybe you could, handle the case if there is no "." in filename/filepath

Comment (friewerts) [reply packages/webpack-config/src/helpers/getFileExtension/getFileExtension.js:8]: we should also have a list of possible filetypes (js, jsx, ts, tsx) and throw a warning/error if non of them fits

File: packages/webpack-config/src/helpers/getFileExtension/getFileExtension.test.js [hunk 8]
```diff
@@ -0,0 +1,42 @@
+import getFileExtension from "./getFileExtension";
```
Comment (friewerts) [inline packages/webpack-config/src/helpers/getFileExtension/getFileExtension.test.js:1]: the mentioned edge-cases could also be part of the test

File: packages/webpack-config/src/helpers/paths.js [hunk 9]
```diff
@@ -64,20 +67,25 @@ export const appPath = resolveApp(".");
 export const appSrc = resolveApp(config.srcPath);
 
 export const jsEntry = () => {
-  if (config.jsEntry !== defaultConfig.jsEntry) return resolveApp(config.jsEntry);
+  if (config.jsEntry !== defaultConfig.jsEntry) {
```
Comment (friewerts) [inline packages/webpack-config/src/helpers/paths.js:70]: auto-format?

File: packages/webpack-config/src/helpers/paths.js [hunk 10]
```diff
@@ -64,20 +67,25 @@ export const appPath = resolveApp(".");
...
+  }
 
   const extname = path.extname(config.jsEntry);
 
   return resolveApp(config.jsEntry.replace(extname, getJSExtName(config)));
 };
 
 export const jsLegacyEntry = () => {
-  if (config.jsLegacyEntry !== defaultConfig.jsLegacyEntry) return resolveApp(config.jsLegacyEntry);
+  if (config.jsLegacyEntry !== defaultConfig.jsLegacyEntry) {
```
Comment (friewerts) [inline packages/webpack-config/src/helpers/paths.js:80]: auto-format?

File: packages/webpack-config/babel.config.js (hunk 4 above)
Comment (friewerts) [reply packages/webpack-config/babel.config.js:1]: https://github.com/facebook/jest/issues/3112

File: packages/webpack-config/src/helpers/getFileExtension/getFileExtension.js (hunk 7 above)
Comment (ifahrentholz) [reply packages/webpack-config/src/helpers/getFileExtension/getFileExtension.js:8]: this is not an issue since JS handles that gracefully.

File: packages/pv-scripts/package.json (hunk 2 above)
Comment (ifahrentholz) [reply packages/pv-scripts/package.json:22]: Removed

File: packages/pv-stylemark/package.json (hunk 3 above)
Comment (ifahrentholz) [reply packages/pv-stylemark/package.json:25]: Removed

File: packages/webpack-config/src/helpers/paths.js (hunk 9 above)
Comment (ifahrentholz) [reply packages/webpack-config/src/helpers/paths.js:70]: yep - prettier auto-format.

File: packages/webpack-config/src/helpers/paths.js (hunk 10 above)
Comment (ifahrentholz) [reply packages/webpack-config/src/helpers/paths.js:80]: yep - prettier auto-format.

File: packages/webpack-config/jest.config.js (hunk 5 above)
Comment (ifahrentholz) [reply packages/webpack-config/jest.config.js:1]: Moved the jest-config to root
